core/src/Cuda/Kokkos_Cuda_Instance.hpp

```diff
@@ -193,6 +193,26 @@ class CudaInternal {
     }
   }
 
+  ///////////////////////////////////////////////////////////////////////////////////////////
+
+  template <typename... InputArgs>
+  void cuda_api_interface_safe_call(
+      cudaError_t (*cuda_api_function)(InputArgs...),
+      InputArgs... cuda_api_input) const {
```

Comment:
I think I would use perfect forwarding here, to make this future-proof ;) (sorry to interrupt, I jumped here from #6008)

Reply:
It's a C interface being called down to, so nothing will be large or have constructors or destructors. In general, you'd be right, though.